velox/exec/Window.cpp

```diff
@@ -601,13 +663,14 @@ vector_size_t Window::callApplyLoop(
           partitionOffset_ + numOutputRowsLeft,
           resultIndex,
           result);
+      processedRowCount = numOutputRowsLeft;
       numOutputRowsLeft = 0;
       break;
     }
   }
 
   // Return the number of processed rows.
-  return numOutputRows - numOutputRowsLeft;
+  return {numOutputRows - numOutputRowsLeft, processedRowCount};
```

Comment:
processedRowCount seems unnecessary. You always want to flush out the rows that were processed.

Reply:
@aditi-pandit We need processedRowCount  here. Assuming that the value of numOutputRows is 4096, and there are two partitions. The numRows of the first partition, partition1, is 2046. Since 2046 is less than 4096, the program will enter the if branch and carry out the corresponding operations. After completion, numOutputRowsLeft will be updated to 2050. The numRows of the second partition, partition2, is 3028, which is greater than 2050, so the program will proceed to the else branch. In this branch, numOutputRowsLeft will be set to 0. This will result in the final returned processed row count being 4096, instead of the 3028 rows from partition2. This issue may cause errors during the final execution of clearOutput.

Reply:
Ah... I understand what is going on now.

We could simplify this. The currentPartition_->clearOutputRows can be done in this loop itself after every callApplyForPartitionRows instead of after callApplyLoop.